src/lnp/presentation/encoding.rs

```diff
@@ -395,6 +395,24 @@ mod byte_strings {
         }
     }
 
+    impl LightningEncode for [u8; 3] {
+        fn lightning_encode<E: io::Write>(
+            &self,
+            mut e: E,
+        ) -> Result<usize, io::Error> {
+            e.write_all(self)?;
+            Ok(self.len())
+        }
+    }
+
+    impl LightningDecode for [u8; 3] {
+        fn lightning_decode<D: io::Read>(mut d: D) -> Result<Self, Error> {
+            let mut ret = [0u8; 3];
+            d.read_exact(&mut ret)?;
+            Ok(ret)
+        }
+    }
+
```

Comment:
Not desirable, very edge case. We need to do this only either via newtypes — or inline withing `ShortChannelId` encoding/decoding code

Reply:
This is also used here

https://github.com/LNP-BP/rust-lnpbp/blob/5f1a9d6fce359207c9bad24dbc48dd25ac3890a4/src/lnp/application/message.rs#L719-L724

Maybe use seperate structures for `[u8;3]`?

Reply:
We don't need this for `ShortChannelId` btw. That already has its own special encoding as per BOLT7. I wrote this for above two items. Probably there's something cleaner to handle this. One approach would be to define `rgb_color` and `alias` structure.

Reply:
I think se need to use a newtype for that cases (with Wrapper derive)

Reply:
I am trying with newtype which is wrapper over `[u8;3]`. But the compiler still asking for encoding of `[u8;3]` structures. Which is natural. So it seems we need to specify encoding logic for `[u8;3]` anyway. Whether we use it directly or as a wrapped structure.

Submitting a commit next for better understanding.

Reply:
Pls check my new comments